frame/membership/src/migrations/v4.rs

```diff
@@ -119,26 +112,24 @@ pub fn post_migrate<P: GetStorageVersion, N: AsRef<str>>(old_pallet_name: N, new
 	let new_pallet_name = new_pallet_name.as_ref();
 	log_migration("post-migration", old_pallet_name, new_pallet_name);
 
+	// Assert that nothing remains at the old prefix.
 	let old_pallet_prefix = twox_128(old_pallet_name.as_bytes());
-	#[cfg(test)]
-	{
-		let storage_version_key =
-			[&old_pallet_prefix, &twox_128(STORAGE_VERSION_STORAGE_KEY_POSTFIX)[..]].concat();
-		assert!(storage::next_key(&old_pallet_prefix)
-			.map_or(true, |next_key| !next_key.starts_with(&old_pallet_prefix) ||
-				next_key == storage_version_key));
-	}
-	#[cfg(not(test))]
-	{
-		// Assert that nothing remains at the old prefix
-		assert!(storage::next_key(&old_pallet_prefix)
-			.map_or(true, |next_key| !next_key.starts_with(&old_pallet_prefix)));
-	}
+	let old_pallet_prefix_iter = frame_support::storage::KeyPrefixIterator::new(
+		old_pallet_prefix.to_vec(),
+		old_pallet_prefix.to_vec(),
+		|_| Ok(()),
+	);
+	assert_eq!(old_pallet_prefix_iter.count(), 0);
```

Comment:
is the past  we only checked for the first key at the old prefix, and we accepted that it was the storage_version_key, now we ensure nothing remains there.